python/tank/platform/engine.py

```diff
@@ -206,15 +236,78 @@ def get_env(self):
 
     ##########################################################################################
     # properties used by internal classes, not part of the public interface
-    
+
+    def __create_log_handler(self):
+        """
+        Creates a std python logging LogHandler
+        that dispatches all log messages to the
+        :meth:`Engine._emit_log_message()` method in a threadsafe manner.
+
+        :return: :class:`python.logging.LogHandler` object
+        """
+        # use a thin custom wrapper that dispatches to the callback
+
+        # alias 'self' so we can refer to the engine
+        # object from inside our wrapper class
+        engine_instance = self
+
+        # set up logging handler class
+        class _ToolkitEngineHandler(logging.Handler):
+            """
+            Thin log handling wrapper for dispatch back to the engine.
+            """
+            def emit(self, record):
+                """
+                Emit a log message
+
+                :param record: std log record to handle logging for
+                """
+                # for verbosity, add a 'basename' property to the record to
+                # only contain the leaf part of the logging name
+                # tank.session.asset.tk-maya -> tk-maya
+                # tank.session.asset.tk-maya.tk-multi-publish -> tk-multi-publish
+                record.basename = record.name.split(".")[-1]
+
+                # dispatch all messages to callback and make sure callback always executes
+                # in the main thread. For whether to output debug logging, look at the
+                # menu-based toggle as well as an external environment variable
+                if record.levelno > logging.DEBUG or \
+                        constants.ENGINE_DEBUG_LOGGING_ENV_VAR in os.environ or \
+                        engine_instance.get_setting("debug_logging", False) or \
+                        engine_instance._display_debug:
```

Comment:
Would it be possible to use the handler's setLevel rather than doing the comparison at this point?  Given how often this might be called for debug output that could be a lot more efficient.